static/app/views/preprod/buildDetails/header/useBuildDetailsActions.tsx

```diff
@@ -95,6 +95,16 @@ export function useBuildDetailsActions({
           if (errorJson.detail) {
             if (typeof errorJson.detail === 'string') {
               errorMessage = errorJson.detail;
+            } else if (errorJson.detail.code === 'staff-required') {
+              addErrorMessage(
+                t(
+                  'Re-authenticate as staff first and then return to this page and try again. Redirecting...'
+                )
+              );
+              setTimeout(() => {
+                window.location.href = 'https://sentry.sentry.io/_admin/';
```

Comment:
I think we are meant to use a hook like `useNavigate()` for this sort of thing. Also I wouldn't hardcode `sentry.sentry.io` in here to account for self-hosted admin users.